handlers.go

```diff
@@ -249,7 +250,9 @@ func (dht *IpfsDHT) handlePing(_ context.Context, p peer.ID, pmes *pb.Message) (
 }
 
 func (dht *IpfsDHT) handleFindPeer(ctx context.Context, p peer.ID, pmes *pb.Message) (*pb.Message, error) {
-	defer log.EventBegin(ctx, "handleFindPeer", p).Done()
+	ctx = log.Start(ctx, "handleFindPeer")
+	defer log.Finish(ctx)
```

Comment:
You can actually skip the `SetErr` now and:

1. Set a named error return value (e.g., `_err`). This'll "capture" the err return value in a variable and make it accessible from deferred functions.
2. Use `defer func() { log.FinishWithErr(ctx, _err) }`

But that's up to you.

Reply:
So clever way! Thanks for your idea.